plugin/trino-phoenix5/src/test/java/io/trino/plugin/phoenix5/TestPhoenixConnectorTest.java

```diff
@@ -307,6 +317,50 @@ public void testMissingColumnsOnInsert()
         assertQuery("SELECT * FROM test_col_insert", "SELECT 1, 'val1', 'val2'");
     }
 
+    @Test
+    public void testTopNPushdown()
+    {
+        assertThat(query("SELECT orderkey FROM orders ORDER BY orderkey LIMIT 10"))
+                .ordered()
+                .isNotFullyPushedDown(TopNNode.class, ExchangeNode.class, ExchangeNode.class, LimitNode.class);
+    }
+
+    @Test
+    public void testUseSortedProperties()
+    {
+        String tableName = "test_propagate_table_scan_sorting_properties";
+        String createTableSql = format("" +
+                        "CREATE TABLE %s AS " +
+                        "SELECT * FROM tpch.tiny.customer",
+                tableName);
+        assertUpdate(createTableSql, 1500L);
+
+        String expected = "SELECT custkey FROM customer ORDER BY 1 NULLS FIRST LIMIT 100";
+        String actual = format("SELECT custkey FROM %s ORDER BY 1 NULLS FIRST LIMIT 100", tableName);
+        Session session = getSession();
+        assertQuery(session, actual, expected, assertPartialLimitWithPreSortedInputsCount(session, 1));
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    private Consumer<Plan> assertPartialLimitWithPreSortedInputsCount(Session session, int expectedCount)
```

Comment:
Copied from TestHiveConnector. If we agree this is right I'll extract.

Reply:
Test looks ok

Reply:
@lhofhansl Do you want to extract this somewhere common in an earlier commit?